DB/src/main/java/io/deephaven/db/v2/parquet/ParquetTableWriter.java

```diff
@@ -259,14 +210,24 @@ public static void write(Table t, TableDefinition definition, String path, Map<S
     }
 
     private static Table pretransformTable(Table t, TableDefinition definition) {
-        List<String> columnsTransform = new ArrayList<>();
-        for (ColumnDefinition column : definition.getColumns()) {
-            if (StringSet.class.isAssignableFrom(column.getDataType())) {
-                columnsTransform.add(column.getName() + " = " + column.getName() + ".values()");
+        List<String> updateViewColumnsTransform = new ArrayList<>();
+        List<String> viewColumnsTransform = new ArrayList<>();
+        for (ColumnDefinition<?> column : definition.getColumns()) {
+            final String colName = column.getName();
+            if (t.hasColumns(colName)) {
+                if (StringSet.class.isAssignableFrom(column.getDataType())) {
+                    updateViewColumnsTransform.add(colName + " = " + colName + ".values()");
+                }
+                viewColumnsTransform.add(colName);
+            } else {
+                viewColumnsTransform.add(colName + " = " + TableTools.nullTypeAsString(column.getDataType()));
```

Comment:
See my suggestion on `nullTypeAsString` about using a `NullSelectColumn`. You'll then need `SourceColumn` for the "colName" case.

Reply:
Code or...